plugins/inputs/tail/tail.go

```diff
@@ -272,18 +310,66 @@ func parseLine(parser parsers.Parser, line string, firstLine bool) ([]telegraf.M
 // for changes, parse any incoming msgs, and add to the accumulator.
 func (t *Tail) receiver(parser parsers.Parser, tailer *tail.Tail) {
 	var firstLine = true
-	for line := range tailer.Lines {
-		if line.Err != nil {
+
+	// holds the individual lines of multi-line log entries.
+	var buffer bytes.Buffer
+
+	var timer *time.Timer
+	var timeout <-chan time.Time
+
+	// The multiline mode requires a timer in order to flush the multiline buffer
+	// if no new lines are incoming.
+	if t.multiline.IsEnabled() {
+		timer = time.NewTimer(t.MultilineConfig.Timeout.Duration)
+		timeout = timer.C
+	}
+
+	channelOpen := true
+
+	for {
+		var line *tail.Line
```

Comment:
Not sure what the compiler does here, but it looks like this is being reallocated every iteration, where we could just be setting it to nil.